server/datastore/mysql/hosts.go

```diff
@@ -727,7 +727,32 @@ func (ds *Datastore) LoadHostByNodeKey(ctx context.Context, nodeKey string) (*fl
 	query := `SELECT * FROM hosts WHERE node_key = ?`
 
 	var host fleet.Host
-	switch err := ds.GetContextTryStmt(ctx, &host, query, nodeKey); {
+	switch err := ds.getContextTryStmt(ctx, &host, query, nodeKey); {
+	case err == nil:
+		return &host, nil
+	case errors.Is(err, sql.ErrNoRows):
+		return nil, ctxerr.Wrap(ctx, notFound("Host"))
+	default:
+		return nil, ctxerr.Wrap(ctx, err, "find host")
+	}
+}
+
+// LoadHostByDeviceAuthToken loads the whole host identified by the device auth token.
+// If the token is invalid it returns a NotFoundError.
+func (ds *Datastore) LoadHostByDeviceAuthToken(ctx context.Context, authToken string) (*fleet.Host, error) {
+	const query = `
+    SELECT
+      h.*
+    FROM
+      host_device_auth hda
+    INNER JOIN
+      hosts h
+    ON
+      hda.host_id = h.id
+    WHERE hda.token = ?`
+
+	var host fleet.Host
+	switch err := sqlx.GetContext(ctx, ds.reader, &host, query, authToken); {
```

Comment:
Any reason to not use `ds.getContextTryStmt` here?

Reply:
I thought it was just created for the heavily-requested statement executed when hosts ping back from osquery? Those new endpoints are manually requested by users of fleet desktop, I don't expect them to be under the same load, but maybe I misunderstand what queries/when we should use the prepared statement cache?

Reply:
Makes sense, agree, eventually use it if we eventually need it :)